docs/api/synchronization_library/barrier.md

```diff
@@ -1,29 +1,41 @@
+---
+grand_parent: API
+parent: Synchronization Library
+nav_order: 2
+---
+
 # `<cuda/std/barrier>`, `<cuda/barrier>`
 
-## Extensions in the `cuda::` namespace
+## Extensions
 
-The class template `barrier` takes an additional scope argument, defaulted to `thread_scope_system`.
+The class template `barrier` takes an additional [thread scope] argument,
+  defaulted to `thread_scope_system`.
 
 ```c++
-// This object is barrier suitable for all threads in the system
-cuda::barrier<> a; 
+// This barrier is suitable for all threads in the system.
+cuda::barrier<cuda::thread_scope_system> a;
 
-// This object has the same type as the previous
-cuda::barrier<cuda::thread_scope_system> b; 
+// These barriers have the same type as the previous one.
+cuda::barrier<> ba;
+cuda::std::barrier<> bb;
 
-// This object is a barrier for threads in the same thread block
-cuda::barrier<cuda::thread_scope_block> c; 
+// This barrier is suitable for all threads in the same thread block.
+cuda::barrier<cuda::thread_scope_block> c;
 ```
 
-The class template `barrier` may also be declared without initialization; a new static member `init` that be used to initialize the object.
+The class template `barrier` may also be declared without initialization; a
+friend function `init` may be used to initialize the object.
 
 ```c++
-__shared__ cuda::barrier<cuda::thread_scope_block> b; // shared memory does not allow initialization
+// Shared memory does not allow initialization.
+__shared__ cuda::barrier<cuda::thread_scope_block> b;
 
-init(&b, 1); // use this free function to initialize this object
+init(&b, 1); // Use this friend function to initialize the object.
 ```
 
-In the `device::` sub-namespace a `__device__` free function is available that provides direct access to the underlying PTX state of a `barrier` object, if its scope is `thread_scope_block` and it is allocated in shared memory.
```

Comment:
Qualify `device::` as `cuda::device::` if you want to avoid saying "subnamespace".

Reply:
It's a bad phrase anyways, blame @ogiroux.